service/src/main/java/com/kodemon/service/implementations/TrainerServiceImpl.java

```diff
@@ -140,6 +144,23 @@ public Trainer update(Long id, Trainer trainer) {
     }
 
     @Override
+    public void setFirstSixPokemons(Long trainerId, List<Integer> indices) {
+        List<Pokemon> newActivePokemons = new ArrayList<>();
+        Trainer trainer = findById(trainerId);
+        List<Pokemon> trainersPokemons = trainer.getPokemons();
+
+        for(int index : indices) {
+            Pokemon toBeActive = trainersPokemons.get(index);
```

Comment:
Oh yes, accordin to the documentation, it throws an exception. You can handle it in the try/catch, if you want to. It will make the code ugly, but safer. I know that we have things like this in many places. Nevermind :smile: . Up to you, whether you fix this or not